### PR TITLE
Add timeout

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
As per subject.

### CODE REVIEW

**Comment by jackpal** [comment]
Thanks for the updated pull request!

I'm sorry to cause you trouble, but I did some cleanup work that caused a conflict with your pull request.

(I did incorporate your "Libraries should not log to stdout" change.)

...so if I could impose on you for one more refactoring, would you please:

sync to the latest version of this repo.
Remove the NewClientForDefaultGateway-related API change (because that API is now gone.)

**Comment by AudriusButkevicius** [comment]
Amended.

File: natpmp.go [hunk 1]
```diff
@@ -20,17 +21,23 @@ const RECOMMENDED_MAPPING_LIFETIME_SECONDS = 3600
...
+	timeout time.Duration
 }
 
 // Create a NAT-PMP client for the NAT-PMP server at the gateway.
 func NewClient(gateway net.IP) (nat *Client) {
-	return &Client{&network{gateway}}
+	return &Client{&network{gateway}, 0}
+}
+
+// Create a NAT-PMP client for the NAT-PMP server at the gateway, with a timeout.
```
Comment (jackpal) [inline natpmp.go]: Could you add more text to this comment, explaining what the timeout does, and how it compares to the default NewClient() behavior? (I can kind of figure it out from reading the implementation, but I think the comment should explain it well enough that the caller doesn't have to read the implementation.)

Comment (AudriusButkevicius) [inline natpmp.go]: done.

**Comment by jackpal** [comment]
Thanks!

**Pull request merged**